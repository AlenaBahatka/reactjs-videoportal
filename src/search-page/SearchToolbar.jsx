import React, { PureComponent } from 'react';
import FilterOption from './FilterOption';

class SearchToolbar extends PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            sortTypes: this.props.sortTypes,
            selectedSortType: this.props.selectedSortType,
            numberOfFilms: this.props.numberOfFilms
        }
        this.handleOptionChange = this.handleOptionChange.bind(this);
    }

    handleOptionChange(checkedOption) {
        this.setState({
            selectedSortType: checkedOption
        });
        this.props.sortHandler(checkedOption);
    }
    
    render() {
        return (
            <div className="row">
                <div className="col-sm-9">{this.state.numberOfFilms} movies found</div>
                <div className="col-sm-3">Sort by
                    <div className="btn-group btn-group-lg">
                        {this.state.sortTypes.map((sortType) => 
                            <FilterOption 
                                checkedOption={this.state.selectedSortType} 
                                name={sortType.id}
                                displayName={sortType.displayName}
                                handleOptionChange={this.handleOptionChange}
                                optionName='sort'
                                key={sortType.id}>
                            </FilterOption>
                        )}
                    </div>
                </div>                            
            </div>
        );
    }
}

export default SearchToolbar;