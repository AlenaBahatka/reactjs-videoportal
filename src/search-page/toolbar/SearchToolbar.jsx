import React, { PureComponent } from 'react';
import FilterOption from '../common-components/FilterOption';

class SearchToolbar extends PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			sortTypes: this.props.sortTypes,
			selectedSortType: this.props.selectedSortType
		};
		this.handleOptionChange = this.handleOptionChange.bind(this);
	}

	handleOptionChange(checkedOption) {
		this.setState({
			selectedSortType: checkedOption
		});
		this.props.sortHandler(checkedOption);
	}

	render() {
		const numberOfFilms = this.props.numberOfFilms;
		const movieWord = numberOfFilms === 1 ? 'movie' : 'movies';
		return (
			<div className="row">
				<div className="col-sm-9">
					{this.props.numberOfFilms} {movieWord} found
				</div>
				<div className="col-sm-3">
					Sort by
					<div className="btn-group btn-group-lg">
						{this.state.sortTypes.map((sortType) => (
							<FilterOption
								checkedOption={this.state.selectedSortType}
								name={sortType.id}
								displayName={sortType.displayName}
								handleOptionChange={this.handleOptionChange}
								optionName="sort"
								key={sortType.id}
							/>
						))}
					</div>
				</div>
			</div>
		);
	}
}


export default SearchToolbar;
