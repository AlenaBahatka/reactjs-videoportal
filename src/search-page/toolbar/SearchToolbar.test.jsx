import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchToolbar from './SearchToolbar';

let sortTypes = [
	{
		id: 'releaseDate',
		displayName: 'release date'
	},
	{
		id: 'rating',
		displayName: 'rating'
	}
];
let selectedSortType = 'releaseDate';
let searchToolbar, sortHandlerMock;
describe('SearchToolbar', () => {
	beforeEach(() => {
		sortHandlerMock = jest.fn();

		searchToolbar = (
			<SearchToolbar
				sortTypes={sortTypes}
				selectedSortType={selectedSortType}
				numberOfFilms={1}
				sortHandler={sortHandlerMock}
			/>
		);
	});

	it('should correctly render component in case one film', () => {
		let searchToolbarComponent = shallow(searchToolbar);
		expect(searchToolbarComponent).toMatchSnapshot();
	});

	it('should correctly render component in case some films', () => {
		searchToolbar = (
			<SearchToolbar
				sortTypes={sortTypes}
				selectedSortType={selectedSortType}
				numberOfFilms={3}
				sortHandler={sortHandlerMock}
			/>
		);
		let searchToolbarComponent = shallow(searchToolbar);
		expect(searchToolbarComponent).toMatchSnapshot();
	});

	it('should call sort handler', () => {
		let searchToolbarComponent = mount(searchToolbar);

		searchToolbarComponent.find("input[value='rating']").simulate('change');

		expect(sortHandlerMock).toBeCalled();
	});
});
