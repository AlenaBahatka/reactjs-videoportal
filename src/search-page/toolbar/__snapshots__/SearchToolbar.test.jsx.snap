// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchToolbar should correctly render component in case one film 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9"
  >
    1
     
    movie
     found
  </div>
  <div
    className="col-sm-3"
  >
    Sort by
    <div
      className="btn-group btn-group-lg"
    >
      <FilterOption
        checkedOption="releaseDate"
        displayName="release date"
        handleOptionChange={[Function]}
        key="releaseDate"
        name="releaseDate"
        optionName="sort"
      />
      <FilterOption
        checkedOption="releaseDate"
        displayName="rating"
        handleOptionChange={[Function]}
        key="rating"
        name="rating"
        optionName="sort"
      />
    </div>
  </div>
</div>
`;

exports[`SearchToolbar should correctly render component in case some films 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9"
  >
    3
     
    movies
     found
  </div>
  <div
    className="col-sm-3"
  >
    Sort by
    <div
      className="btn-group btn-group-lg"
    >
      <FilterOption
        checkedOption="releaseDate"
        displayName="release date"
        handleOptionChange={[Function]}
        key="releaseDate"
        name="releaseDate"
        optionName="sort"
      />
      <FilterOption
        checkedOption="releaseDate"
        displayName="rating"
        handleOptionChange={[Function]}
        key="rating"
        name="rating"
        optionName="sort"
      />
    </div>
  </div>
</div>
`;
